### PR TITLE
Additional form fields

### LINKED ISSUE #12: 12. As a user, I want to input details about the material I have.
## Notes
The specificity of the location options needs to be determined. This will depend on what data will be most useful to the team.

## Acceptance criteria
The form will include:
- [x] primary material
- [x] primary material percentage
- [x] secondary material
- [x] secondary material percentage
- [x] material source (post consumer or post industrial)
- [x] weight (lbs./kg)
- [x] user location
- [x] if user is a company or an individual
- [x] add is required to applicable fields

### PR DESCRIPTION
**Pull request by hana-acciaioli**
### Additional form fields added:

- weight
- mass units
- postal code
- user group

### Other edits:

- Added "required" attributes to some form fields, making it necessary to pass in FormEvent type to the "handlesubmit" function.
- Added "autofocus" attribute to first form field

## Related Issue
closses #12 

### Before
image

### After
image

## Acceptance Criteria

- [x]  primary material percentage
- [x]  secondary material
- [x]  secondary material percentage
- [x]  material source (post consumer or post industrial)
- [x]  weight (lbs./kg)
- [x]  user location
- [x]  if user is a company or an individual
- [x]  add is required to applicable fields

## Testing Steps / QA Criteria

1. Click **Submit** without inputting any information. Expect to see alert preventing form submission and directing you to select a primary material.
2. Add the **Primary Material** and click **Submit**. Expect to see alert preventing form submission and directing you to select a weight.
3.  Add a **wieght** and click **Submit**. Expect to see alert preventing form submission and directing you to select a Material Source.
4. Add the **Material Source** and  click **Submit**. Expect to see alert preventing form submission and directing you to input a postal code.
5. Input the **Postal Code Location** and click **Submit**. Expect to see alert preventing form submission and directing you to select a user group.
6. Select the **User Group** and click **Submit**. Expect to see a list of recyclers that match with the material information you input.

### CODE REVIEW

File: src/components/MaterialFilter.tsx [hunk 1]
```diff
@@ -114,7 +120,7 @@ const MaterialFilter: React.FC = () => {
                   placeholder="Secondary Material"
```
Comment (emilysellers) [inline src/components/MaterialFilter.tsx:120]: It'd probably be helpful to remove the placeholder text here for accessibility purposes. We could take care of that later though.

File: src/components/MaterialFilter.tsx [hunk 2]
```diff
@@ -137,27 +143,67 @@ const MaterialFilter: React.FC = () => {
...
-        {/* <label>
+        <label>
           Weight
-          <input className="filter-select" placeholder="Weight" id="weight"></input>
-        </label> */}
+          <input
+            type="number"
+            className="filter-select"
+            id="weight"
+            placeholder="Weight"
```
Comment (emilysellers) [inline src/components/MaterialFilter.tsx]: Again, probably best to remove placeholder text.

File: src/components/MaterialFilter.tsx [hunk 3]
```diff
@@ -137,27 +143,67 @@ const MaterialFilter: React.FC = () => {
...
-            </select>
-          </label>
-          {/* I am a...
-          <select>
+          Material Postal Code Location
+          <input
+            type="text"
+            className="filter-select"
+            id="zip"
+            placeholder="Zip Code"
```
Comment (emilysellers) [inline src/components/MaterialFilter.tsx]: And one more placeholder text to remove.

**Review by emilysellers — APPROVED** [review]
This looks great! Love the autofocus and the pop-up alerts. I added a couple of comments about removing placeholder text - we could take care of that later though. And just checking: this PR implements the form inputs but we'll still need to work out the query logic for the weight value, correct?

File: src/components/MaterialFilter.tsx (hunk 1 above)
Comment (hana-acciaioli) [reply src/components/MaterialFilter.tsx:120]: Good call! I'll do that now.

File: src/components/MaterialFilter.tsx (hunk 2 above)
Comment (hana-acciaioli) [reply src/components/MaterialFilter.tsx]: Removed `placeholder = "weight"`

File: src/components/MaterialFilter.tsx (hunk 3 above)
Comment (hana-acciaioli) [reply src/components/MaterialFilter.tsx]: Removed `placeholder="Zip Code"`

**Comment by hana-acciaioli** [comment]
@emilysellers, because the data does not currently have weights, we cannot use it as a query parameter yet. However, we will be using this information to learn more about the users  and will be storing the inputs in our (future) user_data table.

**Pull request merged**